packages/menu/src/menu.vue

```diff
@@ -50,6 +50,7 @@ import {
 import Menubar from '@element-plus/utils/menu/menu-bar'
 import ElMenuCollapseTransition from './menu-collapse-transition.vue'
 import useMenuColor from './useMenuColor'
+import { isNavigationFailure, NavigationFailureType } from 'vue-router'
```

Comment:
Combination of other library could be a burden on Element-Plus. Please remove this dependency and change the implementation.

Reply:
Agreed, I had this doubt at the beginning of reviewing this code, but I am not quite sure since the `vue-router` is not actually coupled with this component, that was why we fetched the object from `appContext` instead of importing it directly.

Even with tree-shaking I don't think that importing members from `vue-router` is a good idea. We should definitely put a line of comment here to fill in the context.